src/components/AuthorViewpoint/FactListCard.tsx

```diff
@@ -0,0 +1,117 @@
+"use client";
+import { Fact } from "@/types/conversations.types";
+import EditViewpointFact from "@/components/AuthorViewpoint/EditViewpointFact";
+import { MagnifyingGlassIcon, PlusIcon } from "@heroicons/react/24/outline";
+import { Select, Button } from "@mantine/core";
+import { useState, useEffect, Dispatch, SetStateAction } from "react";
+import { allFacts } from "@/mock/conversationMock";
+
+type FactListCardProps = {
+    viewpointFactList: Fact[];
+    setViewpointFactList: Dispatch<SetStateAction<Fact[]>>;
+};
+
+export default function FactListCard({
+    viewpointFactList,
+    setViewpointFactList,
+}: FactListCardProps) {
+    const [searchData, setSearchData] = useState<Fact[]>(allFacts);
+    const [selectedFactId, setSelectedFactId] = useState<string>("");
+    const [searchValue, setSearchValue] = useState<string>(""); // eslint-disable-line
+
+    useEffect(() => {
+        const selectedFact = searchData.find(
+            (fact) => String(fact.id) === selectedFactId,
+        );
+        if (selectedFact) {
+            //move the selected fact to the viewpointFactList
+            setViewpointFactList((prev) => [...prev, selectedFact]);
+
+            //remove the selected fact from the searchData
+            const newSearchData = searchData.filter(
+                (fact) => fact.id !== selectedFact.id,
+            );
+            setSearchData(newSearchData);
+            setSelectedFactId("");
+        }
+    }, [
+        selectedFactId,
+        setSelectedFactId,
+        viewpointFactList,
+        setViewpointFactList,
+        searchData,
+    ]);
+
+    const removeFact = (factId: string) => {
+        //remove the fact from the viewpointFactList
+        const removedFact = viewpointFactList.find(
+            (fact) => String(fact.id) == factId,
+        );
+        setViewpointFactList(
+            viewpointFactList.filter((fact) => String(fact.id) !== factId),
```

Comment:
This again may cause race conditions since state may not be updated immideiatly. It is recommended that both the `find` and `filter` manipulation be put inside the setState arrow function.